SYCL/SubGroup/sub_group_as.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fno-sycl-id-queries-fit-in-int %s -o %t.out
```

Comment:
I wonder, why those changes are needed at all?

Reply:
Thanks @AlexeySachkov for reviewing the patch. 

https://github.com/intel/llvm/pull/3427: Enables the option -fsycl-id-queries-fit-in-int to true by default.  sub_group_as.cpp passes when the option is false on default mode but fails with default ON. So I added -fno-sycl-id-queries-fit-in-int to disable the option and match with the old behavior. 

 sycl-id-queries-fit-in-int  is a big performance improvement for DPC++ in many cases. The test failures currently block https://github.com/intel/llvm/pull/3427. Is there any workaround i can use here? Thanks.

Reply:
@smanna12, please create issue for the regression. It looks like the test start to fail after changing default behavior.
In short: load/store functions for local pointer stop work as expected.

Reply:
@AlexeySachkov, could you please review this?

Reply:
It looks like root cause for failing test is the same with post-commit failures after merging intel/llvm#3427

Reply:
> It looks like root cause for failing test is the same with post-commit failures after merging [intel/llvm#3427](https://github.com/intel/llvm/pull/3427)

@vladimirlaz, Does it mean that the test needs to be updated? Not sure i understand your comments.

Reply:
@vladimirlaz, PR: intel/llvm#3427: which has not been merged yet. I am waiting for this PR to be merged first before merging PR: intel/llvm#3427